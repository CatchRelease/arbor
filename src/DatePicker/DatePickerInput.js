import React from 'react';
import PropTypes from 'prop-types';
import DayPickerInput from 'react-day-picker/DayPickerInput';

import datePickerDefaults from './datePickerDefaults';
import DatePickerInputOverlay from './DatePickerInputOverlay';
import Input from '../Input/Input';

const DatePickerInput = ({ dayPickerProps, ...props }) => (
  <DayPickerInput
    component={Input}
    overlayComponent={DatePickerInputOverlay}
    dayPickerProps={{
      ...datePickerDefaults,
      ...dayPickerProps
    }}
    {...props}
  />
);

DatePickerInput.propTypes = {
  dayPickerProps: PropTypes.object // eslint-disable-line react/forbid-prop-types
};

DatePickerInput.defaultProps = {
  dayPickerProps: {}
};

export default DatePickerInput;
