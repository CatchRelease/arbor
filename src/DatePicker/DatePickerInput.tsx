import { FC } from 'react';
import { DayPickerInputProps } from 'react-day-picker';
import DayPickerInput from 'react-day-picker/DayPickerInput';

import datePickerDefaults from './datePickerDefaults';
import DatePickerInputOverlay from './DatePickerInputOverlay';
import Input, { InputProps } from '../Input';

const defaultInputProps = {
  autoComplete: 'off'
};

type Props = Omit<DayPickerInputProps, 'inputProps'> & {
  inputProps?: InputProps;
};

const DatePickerInput: FC<Props> = ({
  dayPickerProps = {},
  inputProps = {},
  ...props
}) => (
  <DayPickerInput
    component={Input}
    overlayComponent={DatePickerInputOverlay}
    dayPickerProps={{
      ...datePickerDefaults,
      ...dayPickerProps
    }}
    inputProps={{
      ...defaultInputProps,
      ...inputProps
    }}
    {...props}
  />
);

export default DatePickerInput;
