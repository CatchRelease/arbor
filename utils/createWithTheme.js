import React from 'react';
import { CacheProvider } from '@emotion/core';
import createCache from '@emotion/cache';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';
import theme from '../src/theme';

const myCache = createCache();
myCache.compat = true;

export default (Component) =>
  create(
    <CacheProvider value={myCache}>
      <ThemeProvider theme={theme}>{Component}</ThemeProvider>
    </CacheProvider>
  );
