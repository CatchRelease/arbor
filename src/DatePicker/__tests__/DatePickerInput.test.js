import React from 'react';
import { shallow } from 'enzyme';

import createWithTheme from '../../../utils/createWithTheme';
import DatePickerInput from '../DatePickerInput';

describe('<DatePickerInput />', () => {
  it('properly renders a DatePicker', () => {
    const tree = createWithTheme(
      <DatePickerInput foo="bar" inputProps={{ id: 'date-picker-input' }} />
    );

    expect(tree).toMatchSnapshot();
  });

  describe('props', () => {
    it('passes props to the DayPickerInput', () => {
      const datePickerInput = shallow(
        <DatePickerInput foo="bar" inputProps={{ id: 'date-picker-input' }} />
      );

      expect(datePickerInput.find('DayPickerInput')).toHaveProp({
        foo: 'bar'
      });
    });

    it('passes dayPickerProps into the dayPickerProps prop', () => {
      const datePickerInput = shallow(
        <DatePickerInput
          dayPickerProps={{ foo: 'bar' }}
          inputProps={{ id: 'date-picker-input' }}
        />
      );

      expect(
        datePickerInput.find('DayPickerInput').props().dayPickerProps.foo
      ).toEqual('bar');
    });

    it('allows overriding autoComplete', () => {
      const tree = createWithTheme(
        <DatePickerInput
          foo="bar"
          inputProps={{ id: 'date-picker-input', autoComplete: 'on' }}
        />
      );

      expect(tree).toMatchSnapshot();
    });
  });
});
