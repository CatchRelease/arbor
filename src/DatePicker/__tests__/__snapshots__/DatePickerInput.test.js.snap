// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePickerInput /> properly renders a DatePicker 1`] = `
.emotion-2 {
  margin-bottom: 1rem;
}

.emotion-0 {
  border-radius: 3px;
  border: 1px solid #E0E1E2;
  box-sizing: border-box;
  color: #25272D;
  font-family: Motiva Sans;
  font-size: 0.875rem;
  line-height: 1.25;
  margin: 0.25rem 0;
  padding: 0.5rem;
  width: 100%;
}

@media (max-width:768px) {
  .emotion-0 {
    font-size: 1rem;
  }
}

.emotion-0::-webkit-input-placeholder {
  color: #989A9C;
}

.emotion-0:-moz-placeholder {
  color: #989A9C;
}

.emotion-0::-moz-placeholder {
  color: #989A9C;
}

.emotion-0:-ms-input-placeholder {
  color: #989A9C;
}

.emotion-0:focus {
  border: 1px solid #5577FF;
  box-shadow: 0 0 8px rgba(85,119,255,0.1);
  outline: none;
}

.emotion-0:disabled {
  background: #F2F2F2;
}

.emotion-0:required {
  box-shadow: none;
}

<div
  className="DayPickerInput"
>
  <div
    className="emotion-2 emotion-3"
  >
    
    <input
      className="emotion-0 emotion-1"
      id="date-picker-input"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      placeholder="YYYY-M-D"
      type="text"
      value=""
    />
    
  </div>
</div>
`;
